public/admin_account.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+include_once '../includes/config.php';
+include '../includes/db_api.php';
+
+
+session_start();
+
+if (!isset($_SESSION['username']) || $_SESSION['role_id_role'] === 0) {
+    header("Location: login.php");
+    exit();
+  }
+  
+  // Retrieve logged-in admin's username from session
+  $username = $_SESSION['username'];
```

Comment:
Na linha 9 permites também que o `role_id_role` seja `0`. Se assim for mas o `username` não esteja definido, aqui vai falhar.